checker/src/org/checkerframework/checker/index/lowerbound/LowerBoundAnnotatedTypeFactory.java

```diff
@@ -261,12 +269,30 @@ public Void visitUnary(UnaryTree tree, AnnotatedTypeMirror typeDst) {
                 case POSTFIX_DECREMENT:
                     // Do nothing. The CF should take care of these itself.
                     break;
+                case BITWISE_COMPLEMENT:
+                    handleBitWiseComplement(
+                            getSearchIndexAnnotatedTypeFactory()
+                                    .getAnnotatedType(tree.getExpression()),
+                            typeDst);
+                    break;
                 default:
                     break;
             }
             return super.visitUnary(tree, typeDst);
         }
 
+        /**
+         * Special case for bitwise complement on the result of binary search that has been compared
+         * to zero.
```

Comment:
I would be a bit more specific:  if the type of `expr` is `@NegativeIndexFor`, then the type of `~expr` is `@IndexFor`.
That's not quite what this routine does -- explaining why would be helpful in understanding the code and how it interacts with other parts of the checker.